jcommune-service/src/main/java/org/jtalks/jcommune/service/security/AclGroupPermissionEvaluator.java

```diff
@@ -340,10 +341,14 @@ private boolean isRestrictedForGroup(List<GroupAce> controlEntries,
     private boolean isGrantedForGroup(List<GroupAce> controlEntries,
                                       Authentication authentication, Permission permission,
                                       boolean isCheckAllowedGrant) {
-        if (authentication.getPrincipal() instanceof JCUser) {
+        Object auth = authentication.getPrincipal();
+        if (auth instanceof JCUser) {
+            List<Long> groupsIDs = ((JCUser) auth).getGroupsIDs();
```

Comment:
I also thought to use HashSet here, but 99% of users are members of only one group, and controlEntries size not so large because it has been filtered previously.